smithy-trait-codegen/src/test/java/software/amazon/smithy/traitcodegen/TraitCodegenPluginTest.java

```diff
@@ -163,4 +93,43 @@ public void doesNotFormatContentInsideHtmlTags() {
                           + "    }\n";
         assertTrue(fileStringOptional.get().contains(expected));
     }
+
+    @Test
+    public void rendersExternalDocumentation() {
+        MockManifest manifest = runTraitCodegen(Collections.emptyList(), Collections.emptyList());
+
+        assertFalse(manifest.getFiles().isEmpty());
+        assertEquals(EXPECTED_NUMBER_OF_FILES, manifest.getFiles().size());
+        Optional<String> fileStringOptional = manifest.getFileString(
+                Paths.get("com/example/traits/structures/StructureTrait.java").toString());
+        assertTrue(fileStringOptional.isPresent());
+        String expectedStructureExternalDocs = "<a href=\"https://external.docs/structureTrait\">External docs</a>";
+        assertTrue(fileStringOptional.get().contains(expectedStructureExternalDocs));
+//        String expectedFieldExternalDocs = "<a href=\"https://external.docs/structureTrait#fieldA\">External docs</a>";
+//        assertTrue(fileStringOptional.get().contains(expectedFieldExternalDocs));
```

Comment:
Made these just for demonstration purposes, can revert as this is not related to this PR. Uncommenting these lines will make the test fail, even though there is 

```
    @externalDocumentation(
        "External docs": "https://external.docs/structureTrait#fieldA"
    )
    fieldA: String
```.

To make it work, it has to contain a documentation section:
```

```
    /// some other docs
    @externalDocumentation(
        "External docs": "https://external.docs/structureTrait#fieldA"
    )
    fieldA: String
```

I believe this is how the interceptors are implemented, the `ExternalDocsInterceptor` has  
```
    @Override
    public Class<JavaDocSection> sectionType() {
        return JavaDocSection.class;
    }
```
but this condition does not hold in the failing case.

I can raise an issue if you also feel that this is a problem